### PR TITLE
Private Eyes only diff

### PR DESCRIPTION
**Pull request by jancajthaml**
minor formatting fixes, minor reduction of if-else with pattern matching instead, earlier pattern matching with conditions in match, minor comments about usage of futures

### CODE REVIEW

File: src/test/scala/play/api/cache/redis/util/ExpirationSpec.scala [hunk 1]
```diff
@@ -4,6 +4,7 @@ import java.util.Date
 
 import play.api.cache.redis.{CacheApi, Expiration, ExpirationImplicits, Redis}
 
+// INFO JodaDateTime is deprecated as for 2018 use ZonedDateTime or LocalDateTime
```
Comment (KarelCemus) [inline src/test/scala/play/api/cache/redis/util/ExpirationSpec.scala]: Thanks for your note, I'm aware of it. However, this is a test and then there is pending #108 for tests refactoring, which will include this change.

File: src/test/scala/play/api/cache/redis/configuration/RedisHostSpec.scala [hunk 2]
```diff
@@ -15,7 +15,7 @@ class RedisHostSpec extends Specification {
         | host: localhost
         | port: 6379
         |}
-      """
+      """.stripMargin
```
Comment (KarelCemus) [inline src/test/scala/play/api/cache/redis/configuration/RedisHostSpec.scala]: This adjustment is redundant and pollutes the code. The tests are designed to be minimal and have the best possible readability and appending `stripMargin` would break it. Thus it's added internally by `WithConfiguration` as you can see.

File: src/main/scala/play/api/cache/redis/impl/Builders.scala [hunk 3]
```diff
@@ -37,6 +37,8 @@ object Builders {
   /** converts the future into the value */
   object SynchronousBuilder extends ResultBuilder[ SynchronousResult ] {
 
+    // INFO Await in reality does fiber fork and can result kernel panic if used
+    // recklessly
```
Comment (KarelCemus) [inline src/main/scala/play/api/cache/redis/impl/Builders.scala]: Thanks for your note, I wasn't aware of it. Can you provide some source, e.g., a book or a specification?

Is there any alternative solution to use the Future inside of the synchronous code?

File: src/main/scala/play/api/cache/redis/connector/AkkaSerializer.scala [hunk 4]
```diff
@@ -49,15 +49,13 @@ trait AkkaSerializer {
 private[ connector ] class AkkaEncoder( serializer: Serialization ) {
 
   /** Unsafe method encoding the given value into a string */
-  def encode( value: Any ): String = value match {
-    // null is special case
-    case null => unsupported( "Null is not supported by the redis cache connector." )
+  def encode( value: Any ): String = Option(value) match {
```
Comment (KarelCemus) [inline src/main/scala/play/api/cache/redis/connector/AkkaSerializer.scala]: You are missing the point here and introduce a very major overhead instead. The `null` value should be never present. If so, it's a bug.

Matching of the `value` and `null` is quite efficient as it is a simple test against a null value, which introduces a very low overhead and then we can move to the actual business.

However, you suggest:

- first, call `Option.apply`, which also **includes the test against null**
- second, construct a new instance of `Some`. `None` should be never used.
- third, you need to always check that it is instance of `Some` and then call `unapply` just to access the actual `value` inside the `Option`

This is much bigger overhead without any added value.

File: src/main/scala/play/api/cache/redis/configuration/RedisConfigLoader.scala [hunk 5]
```diff
@@ -15,11 +15,23 @@ private[ configuration ] object RedisConfigLoader {
...
+
+    private def eagerDecision(): Function1[String, String] =
+      path.isEmpty match {
+        case true => (suffix: String) => suffix
+        case _ => (suffix: String) => path + "." + suffix
+      }
+
+    private def delegate: Function1[String, String] = eagerDecision()
+
+    def /( suffix: String ): String = delegate(suffix)
```
Comment (KarelCemus) [inline src/main/scala/play/api/cache/redis/configuration/RedisConfigLoader.scala]: This change significantly reduces the code readability a complicates its maintenance and I don't see any added value without 10 times more SLOC
